src/monitor.c

```diff
@@ -385,10 +385,22 @@ void transform_monitor(struct monitor *m, enum wl_output_transform transform)
 
 void update_monitor_geometries()
 {
+    struct wlr_output_configuration_v1 *config = wlr_output_configuration_v1_create();
+    
     for (int i = 0; i < server.mons->len; i++) {
         struct monitor *m = g_ptr_array_index(server.mons, i);
-        m->geom = *wlr_output_layout_get_box(server.output_layout, m->wlr_output);
+        struct wlr_output_configuration_head_v1 *config_head = wlr_output_configuration_head_v1_create(config, m->wlr_output);
+        struct wlr_box *monitor_box = wlr_output_layout_get_box(server.output_layout, m->wlr_output);
+
+        arrange_layers(m);
+        arrange_monitor(m);
+        config_head->state.enabled = m->wlr_output->enabled;
+        if (monitor_box) {
+			      config_head->state.x = monitor_box->x;
+			      config_head->state.y = monitor_box->y;
+		    }
```

Comment:
what happened to indentation here xD?

Reply:
Oh and please create variables just in time so it is easier to understand what is going on like monitor_box should be created right above this if statement but only if possible (I don't know if this would change the logic here)